core/platform/lf_arduino_support.c

```diff
@@ -33,15 +33,15 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "../platform.h"
 #include "Arduino.h"
 
-instant_t ns_to_microsec(instant_t time) {
-    return (time / 1000) - ((time / 1000) % 4);
-}
-
 /**
- * Pause execution for a number of nanoseconds.
+ * Pause execution for a number of microseconds.
+ *
+ * This function works very accurately in the range 3 microseconds and up to 16383.
+ * We cannot assure that delayMicroseconds will perform precisely for smaller delay-times.
+ * Larger delay times may actually delay for an extremely brief time.
  *
  * @return 0 for success, or -1 for failure. In case of failure, errno will be
- *  set appropriately (see `man 2 clock_nanosleep`).
+ *  set appropriately.
```

Comment:
The code does not set errno.  Also, I don't understand why an argument under 3 microseconds is an error. Why doesn't this just return?

Reply:
I set it to error due to the lack of precision for the delay. The alternative would be to give a warning but not error, which I assume is what you'd like be implemented.

Reply:
The purpose of this function is wait in physical time until physical time closely enough matches logical time. If the precision of the timer is such that it can't wait a small amount of time, then physical time is already closely enough matching logical time. Matching any closer is not possible because of the imprecision of the measurement of time. Hence, I think the function should just return and not issue any warning.

Reply:
Makes sense. Will return as normal.